Sources/MapboxMaps/Foundation/Extensions/Core/MBXGeometry.swift

```diff
@@ -8,81 +8,81 @@ public typealias Geometry = Turf.Geometry
 
 extension MapboxCommon.Geometry {
 
+    /// Allows a `Geometry` object to be initialized with a `Geometry` object.
+    /// - Parameter geometry: The `Geometry` object to transform into the `Geometry` type.
+    internal convenience init(geometry: Geometry) {
+        switch geometry {
+        case .point(let point):
+            let coordinate = point.coordinates
+            self.init(coordinate: coordinate)
+
+        case .lineString(let line):
+            self.init(line: line.coordinates)
+
+        case .polygon(let polygon):
+            self.init(polygon: polygon.coordinates)
+
+        case .multiPoint(let multiPoint):
+            self.init(multiPoint: multiPoint.coordinates)
+
+        case .multiLineString(let multiLine):
+            self.init(multiLine: multiLine.coordinates)
+
+        case .multiPolygon(let multiPolygon):
+            self.init(multiPolygon: multiPolygon.coordinates)
+
+        case .geometryCollection(let geometryCollection):
+            let geometryValues = geometryCollection.geometries.map {( MapboxCommon.Geometry(geometry: $0) )}
+            self.init(geometryCollection: geometryValues)
+
+        #if USING_TURF_WITH_LIBRARY_EVOLUTION
+        @unknown default:
+            fatalError("Could not determine Geometry from given Turf Geometry")
+        #endif
+        }
+    }
+
     /// Initialize a `Geometry` point from a coordinate.
     /// - Parameter coordinate: The coordinate to represent the `Geometry` point.
-    public convenience init(coordinate: CLLocationCoordinate2D) {
+    private convenience init(coordinate: CLLocationCoordinate2D) {
```

Comment:
This PR introduced some compiler errors because some examples were still using this initializer:

```
/path/to/mapbox-maps-ios/Apps/Examples/Examples/All Examples/OfflineRegionManagerExample.swift:49:108: Argument passed to call that takes no arguments
/path/to/mapbox-maps-ios/Apps/Examples/Examples/All Examples/OfflineManagerExample.swift:127:57: Argument passed to call that takes no arguments
```

Reply:
#714